src/prog_models/prognostics_model.py

```diff
@@ -868,21 +871,19 @@ def simulate_to_threshold(self, future_loading_eqn, first_output = None, thresho
         output = self.__output
         thresthold_met_eqn = self.threshold_met
         event_state = self.event_state
+        progress = config['progress']
         if 'thresholds_met_eqn' in config:
             check_thresholds = config['thresholds_met_eqn']
+            threshold_keys = []
         elif threshold_keys is None: 
-            # Note: Dont use implicit boolean in this check- it would then activate for an empty array
-            def check_thresholds(thresholds_met):
-                t_met = thresholds_met.values()
-                if len(t_met) > 0 and not np.isscalar(list(t_met)[0]):
-                    return np.any(t_met)
-                return any(t_met)
-        else:
-            def check_thresholds(thresholds_met):
-                t_met = [thresholds_met[key] for key in threshold_keys]
-                if len(t_met) > 0 and not np.isscalar(list(t_met)[0]):
-                    return np.any(t_met)
-                return any(t_met)
+            # Note: Setting threshold_keys to be all events if it is None
+            threshold_keys = self.events
+
+        def check_thresholds(thresholds_met):
```

Comment:
Oops- I misadvised you on this. This will be a problem if someone uses the thresholds_met_eqn above. It'll be overwritten here. 

My suggested solutions is to move the definition of check_thresholds to above the if 'thresholds_met_eqn' in config: line.